File: src/test/java/com/twsg/lnl/kotlin/c3_extensions/g1_java/CharacterUtilTest.java
package com.twsg.lnl.kotlin.c3_extensions.g1_java;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;
import org.junit.runners.Parameterized.Parameter;
import org.junit.runners.Parameterized.Parameters;

import static org.junit.Assert.assertEquals;

/**
 * @author tonyl
 */
@RunWith(Parameterized.class)
public class CharacterUtilTest {

    @Parameters
    public static Object[][] params() {
        return new Object[][]{
                {'A', true},
                {'a', true},
                {'E', true},
                {'e', true},
                {'U', true},
                {'u', true},
                {'I', true},
                {'i', true},
                {'O', true},
                {'o', true},
                {'B', false},
                {'c', false},
                {'D', false},
        };
    }

    @Parameter(0)
    public char testChar;

    @Parameter(1)
    public boolean expectedVowel;

    @Test
    public void isVowel() throws Exception {
        assertEquals(expectedVowel, CharacterUtil.isVowel(testChar));
    }

}

File: src/test/java/com/twsg/lnl/kotlin/c3_extensions/g2_kotlin/CharacterUtilTest.kt
package com.twsg.lnl.kotlin.c3_extensions.g2_kotlin

import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized
import org.junit.runners.Parameterized.Parameters

/**
 * @author tonyl
 */
@RunWith(Parameterized::class)
class CharacterUtilTest(val testChar: Char, val expectedVowel: Boolean) {

    @Test
    fun isVowel() {
        assertEquals(expectedVowel, testChar.isVowel)
    }

    companion object {

        @Parameters
        @JvmStatic
        fun params(): List<Array<Any>> {
            return listOf(
                    arrayOf('A', true),
                    arrayOf('a', true),
                    arrayOf('E', true),
                    arrayOf('e', true),
                    arrayOf('U', true),
                    arrayOf('u', true),
                    arrayOf('I', true),
                    arrayOf('i', true),
                    arrayOf('O', true),
                    arrayOf('o', true),
                    arrayOf('B', false),
                    arrayOf('c', false),
                    arrayOf('D', false)
            )
        }
    }

}